src/front/src/DocumentPreview.js

```diff
@@ -0,0 +1,44 @@
+import {useEffect, useState} from 'react';
+import {Document, Page, pdfjs} from 'react-pdf';
+import "./DocumentPreview.css";
+
+pdfjs.GlobalWorkerOptions.workerSrc = `//unpkg.com/pdfjs-dist@${pdfjs.version}/build/pdf.worker.min.js`;
+
+/** @type {(document: Blob) => boolean} */
+export const isPdf = document => ['application/pdf', 'application/x-pdf'].includes(document.type);
+
+const plainTextSet = new Set([
+    'bsh',
+    'csh',
+    'ecmascript',
+    'javascript',
```

Comment:
I checked and .js files have a MIME type of text/javascript, so this may be unnecessary, or an indication that other file types may need to be added (e.g. text/x-c should work for C & C++ code, and text/x-java-source for Java code).